pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/GreatestTransformFunction.java

```diff
@@ -44,6 +47,35 @@ public int[] transformToIntValuesSV(ValueBlock valueBlock) {
     return _intValuesSV;
   }
 
+  @Override
+  public Pair<int[], RoaringBitmap> transformToIntValuesSVWithNull(ValueBlock valueBlock) {
+    int numDocs = valueBlock.getNumDocs();
+    initIntValuesSV(numDocs);
+    Pair<int[], RoaringBitmap> values = _arguments.get(0).transformToIntValuesSVWithNull(valueBlock);
+    System.arraycopy(values.getLeft(), 0, _intValuesSV, 0, numDocs);
+    RoaringBitmap nullBitmap = values.getRight();
+    for (int i = 1; i < _arguments.size(); i++) {
+      values = _arguments.get(i).transformToIntValuesSVWithNull(valueBlock);
+      RoaringBitmap curNull = values.getRight();
+      for (int j = 0; j < numDocs & j < values.getLeft().length; j++) {
+        // Ignore null values.
+        if (curNull == null || !curNull.contains(j)) {
+          if (nullBitmap != null && nullBitmap.contains(j)) {
```

Comment:
Rename these variables better, curNull and nullBitmap sound confusing. Maybe `combinedNullBitmap` and `currentNullBitmap`

Reply:
Oops.. This doesn't belong to this PR...